third_party/move/move-compiler-v2/tests/checking/naming/unused_type_parameter_struct.move

```diff
@@ -0,0 +1,11 @@
+module 0x42::test {
+	struct S0<T> {}
+
+	struct S1<T1, T2> {}
+
+	struct S2<T1, phantom T2> {
+		f: S3<T2>,
+	}
+
```

Comment:
Could we add some more tests for coverage here (or in a different test file)?
- use vectors, maybe some more depth of structs, other stuff asked I asked for reachability if they are reachable?
- add recursive structs with type params (to perhaps catch any unwanted re-orderings of passes).

Reply:
done